### ISSUE TITLE
MD007 - Unordered lists nested in ordered lists are treated as a separate list

### ISSUE THREAD

**Issue by sulphi-fox** [DavidAnson/markdownlint#1644]
**What happened?**
`MD007` causes unordered lists that are nested in an ordered list to be treated as an incorrectly indented (separate) list.

**What did I expect to happen?**
I expected it to detect the nested list as correctly indented.

**What am I using?**
I am using the latest version of `markdownlint-cli2` on macOS Sonoma 14.5.

Example:

```md
1. Ordered
  * Unordered
```

Triggers `MD007`: `Unordered list indentation [Expected: 0; Actual: 2]`.

**Comment by DavidAnson** [comment]
The lint message is correct. Because of the insufficient indenting of the second list, it is treated as a second top-level list. You can confirm this in the rendering: https://dlaa.me/markdownlint/#%25m%23%20Issue%201644%0A%0A1.%20Ordered%0A%20%20*%20Unordered%0A

Indenting the second list by three spaces as required to make it part of the first list produces the intended rendering and no warnings as expected: https://dlaa.me/markdownlint/#%25m%23%20Issue%201644%0A%0A1.%20Ordered%0A%20%20%20*%20Unordered%0A

**Comment by sulphi-fox** [comment]
@DavidAnson thanks for the response - I'll close this issue and open another one as a suggestion.
